### PR TITLE
refactor(api): Include `team` and `organization` in serializer context

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

This'd be useful for #13474 and in general. Shouldn't have an impact on the number of Postgres queries, but let's test this by running CI.

Update: This should be good to go. We now include `access_team` and `access_organization` lambdas in serializer context in all viewsets inheriting from `StructuredViewSetMixin`. This allows accessing the relevant team/org in an easy and optimized way (zero extra `SELECT` queries).

### CODE REVIEW

File: posthog/api/cohort.py [hunk 1]
```diff
@@ -96,7 +95,6 @@ def _handle_static(self, cohort: Cohort, request: Request):
 
     def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> Cohort:
         request = self.context["request"]
-        Team.objects.get(pk=self.context["team_id"])
```
Comment (thmsobrmlr) [inline posthog/api/cohort.py:99]: 🤔

File: posthog/api/routing.py [hunk 2]
```diff
@@ -150,7 +150,12 @@ def parents_query_dict(self) -> Dict[str, Any]:
         return result
 
     def get_serializer_context(self) -> Dict[str, Any]:
-        return {**super().get_serializer_context(), **self.parents_query_dict}
+        serializer_context = super().get_serializer_context()
+        serializer_context.update(self.parents_query_dict)
+        # The below are lambdas for lazy evaluation (i.e. we only query Postgres for team/org if actually need)
```
Comment (thmsobrmlr) [inline posthog/api/routing.py:155]:
```suggestion
        # The below are lambdas for lazy evaluation (i.e. we only query Postgres for team/org if actually needed)
```

**Review by thmsobrmlr — APPROVED** [review]
Very cool solution with the lazy evaluation!

Not sure if "access" is a special name in the Python/Django world, but "get_team" would be more obvious to me.

**Comment by Twixes** [comment]
I went for `access` over `get`, as for me "get" _sounds_ costlier than "access", and in this case the thing is that using the function usually does not incur an extra cost. Ideally this would be `@property`-style lazy evaluation, but that's not a thing with dicts, which is the only reason this is a function. Will change back to `get` though.

**Pull request merged**